README.md

```diff
@@ -15,3 +15,9 @@ the function and run automatically when you commit through a GitHub action.
 
 What is the worst-case big $\Theta$ complexity of your implementation? Add your
 answer, including your reasoning, to this markdown file.
+
+I had ai and other github assigmnets to help make the test code, and help with troublshooting the test case of no cycle and test case 3.
+
+The worst cas big theta complexity of my implementation is big theta (n). The overall time complexity is dominated by the number of edges so its big theta (n) where n is the number of edges in the graph. The find and union functions come out to be basically constant because of the compression on the structures. The outer loop iterates over every node in the graph and the inner loop processes them but in constant time making it just the number of edges.
```

Comment:
The overall complexity is dominated by the number of edges because most of the work is processing the edges, but in the special case of there being no edges the runtime would be dominated by the nodes making it big theta (n)  where n is the number of nodes.   The union find operation still initializes the the data structure for all the nodes but no union or find operations are executed.

Reply:
Hmm, sounds like that should somehow be reflected in the complexity.